src/index.ts

```diff
@@ -134,6 +135,7 @@ export default class Api {
   public setBaseUrl() {
     setAxiosDefaults({
       baseURL: `${this.config.settings.https == true ? 'https' : 'http'}://${this.config.settings.host ?? DEFAULT_HOST}`,
+      timeout: this.config.settings.timeout,
```

Comment:
recheck pls if this one works

Reply:
+1 here, have some doubts about it possibly being `undefined`.